File: app/src/main/java/com/example/tetrisapp/ui/viewmodel/LeaderboardViewModel.java
package com.example.tetrisapp.ui.viewmodel;

import androidx.lifecycle.ViewModel;

import com.example.tetrisapp.ui.adapters.ScoresRecyclerViewAdapter;
import com.example.tetrisapp.util.FirebaseTokenUtil;

import java.util.ArrayList;
import java.util.List;

public class LeaderboardViewModel extends ViewModel {
    private String token;
    private final List<ScoresRecyclerViewAdapter.Score> scores = new ArrayList<>();

    public LeaderboardViewModel() {
    }

    public List<ScoresRecyclerViewAdapter.Score> getScores() {
        return scores;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }
}

File: app/src/main/java/com/example/tetrisapp/ui/viewmodel/LeaderboardViewModel.kt
package com.example.tetrisapp.ui.viewmodel

import androidx.lifecycle.*
import com.example.tetrisapp.model.remote.response.PublicRecord

class LeaderboardViewModel : ViewModel() {
    var token: String? = null
    var page = 1
    var pageCount = 1

    private val mutableScores: MutableLiveData<List<PublicRecord>> by lazy {
        MutableLiveData(ArrayList())
    }
    val scores: LiveData<List<PublicRecord>> = mutableScores

    fun updateScores(list: List<PublicRecord>) {
        mutableScores.value = list
    }
}